pulsar-common/src/main/java/org/apache/pulsar/common/util/Runnables.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.util;
+
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class Runnables {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Runnables.class);
+
+    /**
+     * Wraps a Runnable so that throwables are caught and logged when a Runnable is run.
+     *
+     * The main usecase for this method is to be used in {@link java.util.concurrent.ScheduledExecutorService#scheduleAtFixedRate(Runnable, long, long, TimeUnit)}
+     * calls to ensure that the scheduled task doesn't get cancelled as a result of an uncaught exception.
+     *
+     * @param runnable The runnable to wrap
+     * @return a wrapped Runnable
+     */
+    public static Runnable catchingAndLoggingThrowables(Runnable runnable) {
```

Comment:
What about 'safeRun'?
This name sounds too long and verbose to me

Reply:
It's intentionally long and descriptive and does exactly what the method name says. I explained in the PR description why I think SafeRunnable / safeRun is a bad concept.